foreman.te

```diff
@@ -303,6 +294,32 @@ tunable_policy(`foreman_rails_can_connect_all',`
     corenet_tcp_connect_all_ports(foreman_rails_t)
 ')
 
+######################################
+#
+# Logrotate and cron
+#
+
+gen_require(`
+    type logrotate_t;
+    type system_cronjob_tmp_t;
+')
+
+# Allow sending signals in logrotate scripts
+allow logrotate_t foreman_rails_unit_file_t:file read_file_perms;
+allow logrotate_t foreman_rails_unit_file_t:service manage_service_perms;
+
+# RHBZ#1527522: logrotate does not allow exec of systemctl and signals
+# because we do carry some unconfined service logrotation script
+# /etc/logrotate.d/foreman-proxy with logs as var_log_t
+files_list_var(logrotate_t)
+systemd_config_generic_services(logrotate_t)
+
+# When Foreman cronjob is started before Ruby on Rails, /tmp/bundler
+# is created with system_u:object_r:system_cronjob_tmp_t:s0 label denying
+# access to the web process
+manage_files_pattern(foreman_rails_t, system_cronjob_tmp_t, system_cronjob_tmp_t)
+manage_dirs_pattern(foreman_rails_t, system_cronjob_tmp_t, system_cronjob_tmp_t)
+
```

Comment:
During my testing, logrotation was triggered and I saw bunch of denials :-) You should be able to trigger rotation via

    logrotate -vf /etc/logrotate.d/foreman

But I am not sure if this will run in the logrotate domain or not.

Also the manage files/dirs patters are necessary because I ran into special case - when I had my services down, cron was triggered, booted up Rails and it created its temporary directory with a wrong file label. Then service could not be started. This fixes it.